src/pkcs15init/pkcs15-setcos.c

```diff
@@ -349,22 +349,25 @@ setcos_create_key(sc_profile_t *profile, sc_pkcs15_card_t *p15card,
 
 	/* Replace the path of instantiated key template by the path from the object data. */
 	memcpy(&file->path, &key_info->path, sizeof(file->path));
+	if (file->path.len < 2) {
+		LOG_TEST_RET(ctx, SC_ERROR_INVALID_DATA, "Invalid path");
+	}
```

Comment:
Shouldn't the `file` be also freed here?

Reply:
Yes. And the check can be probably moved before the allocation to save the trouble here. Will fix!